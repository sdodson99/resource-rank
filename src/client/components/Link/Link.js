import React from 'react';
import LinkBase from 'next/link';
import PropTypes from 'prop-types';
import useMockContext from '@/hooks/use-mock-context';

const Link = ({ url, children }) => {
  const mock = useMockContext();

  const getHref = () => {
    if (!mock) {
      return url;
    }

    return {
      ...url,
      query: {
        ...url.query,
        mock,
      },
    };
  };

  return <LinkBase href={getHref()}>{children}</LinkBase>;
};

Link.propTypes = {
  url: PropTypes.shape({
    pathname: PropTypes.string,
    query: PropTypes.object,
  }).isRequired,
  children: PropTypes.node,
};

Link.defaultProps = {};

export default Link;
