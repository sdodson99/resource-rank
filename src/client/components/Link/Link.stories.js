import { MockProvider } from '@/hooks/use-mock-context';
import React from 'react';
import Link from './Link';

export default {
  component: Link,
  title: 'Components/Link',
  decorators: [
    (Story) => {
      return;
      <MockProvider>
        <Story />
      </MockProvider>;
    },
  ],
};

const Template = (args) => <Link {...args} />;

export const Default = Template.bind({});
Default.args = {
  url: {
    pathname: '/topics',
  },
  children: 'Click Me',
};
