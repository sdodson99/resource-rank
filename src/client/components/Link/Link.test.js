import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import Link from './Link';
import useMockContext from '@/hooks/use-mock-context';

jest.mock('@/hooks/use-mock-context');

describe('<Link />', () => {
  let props;

  beforeEach(() => {
    props = {
      url: {
        pathname: 'test',
      },
      children: 'Click me',
    };
  });

  afterEach(() => {
    useMockContext.mockReset();
  });

  it('should mount', () => {
    render(<Link {...props} />);

    const link = screen.getByTestId('Link');

    expect(link).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<Link {...props} />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with mock', () => {
    props.url.query = { other: 'value' };
    useMockContext.mockReturnValue('mock1');

    const component = createRenderer().render(<Link {...props} />);

    expect(component).toMatchSnapshot();
  });
});
