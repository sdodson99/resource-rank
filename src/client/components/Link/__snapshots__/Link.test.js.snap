// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> should render correctly 1`] = `
<Link
  href={
    Object {
      "pathname": "test",
    }
  }
>
  Click me
</Link>
`;

exports[`<Link /> should render correctly with mock 1`] = `
<Link
  href={
    Object {
      "pathname": "test",
      "query": Object {
        "mock": "mock1",
        "other": "value",
      },
    }
  }
>
  Click me
</Link>
`;
